src/apps/vpn/notificationhandler.cpp

```diff
@@ -333,10 +339,37 @@ void NotificationHandler::messageClickHandle() {
   }
 
   if (!ExternalOpHandler::instance()->request(
-          ExternalOpHandler::OpNotificationClicked)) {
+          MozillaVPN::OpNotificationClicked)) {
     return;
   }
 
   emit notificationClicked(m_lastMessage);
   m_lastMessage = None;
 }
+
+void NotificationHandler::addonCreated(Addon* addon) {
+  if (addon->type() != "message") {
+    return;
+  }
+
+  if (addon->enabled()) {
+    maybeAddonNotification(addon);
+  }
+
+  connect(addon, &Addon::conditionChanged, this, [this, addon](bool enabled) {
+    if (enabled) {
+      maybeAddonNotification(addon);
+    }
+  });
+}
+
+void NotificationHandler::maybeAddonNotification(Addon* addon) {
+  Q_ASSERT(addon->type() == "message");
```

Comment:
nit: Can this function just receive an AddonMessage instead?

Reply:
no because the AddonManager emits a signal when a new addon is created. We do not have the addon-type filtering. Plus addon-types will be removed soon, right?